pageserver/src/bin/pageserver.rs

```diff
@@ -388,6 +390,9 @@ fn start_pageserver(
             // NOTE: unlike many futures in pageserver, this one is cancellation-safe
             let guard = scopeguard::guard_on_success((), |_| tracing::info!("Cancelled before initial load completed"));
 
+            init_remote_done_rx.wait().await;
+            startup_checkpoint("initial_tenant_load_remote", "Remote part of initial load completed");
```

Comment:
Globally, the only thing waiting for this barrier is the startup_checkpoint() hook -- it's not delaying any work other than the next startup_checkpoint() at the end of the full tenant load.

Locally within tenants there is a behavioral change that they'll wait for all their timelines' remote content to download before starting local load, which I expect will have little practical impact since the number of timelines per tenant is typically small.  One might ideally like to have more of a streaming behavior similar to what we do in attach(), where we use remote metadata per-timeline as soon as it's ready, but I think having a distinct local & remote phase is more desirable for observability and reasoning about the system.